src/node_buffer.cc

```diff
@@ -563,12 +564,15 @@ void Copy(const FunctionCallbackInfo<Value> &args) {
 
 void Fill(const FunctionCallbackInfo<Value>& args) {
   Environment* env = Environment::GetCurrent(args);
+  Local<Context> ctx = env->context();
 
   THROW_AND_RETURN_UNLESS_BUFFER(env, args[0]);
   SPREAD_BUFFER_ARG(args[0], ts_obj);
 
-  size_t start = args[2]->Uint32Value();
-  size_t end = args[3]->Uint32Value();
+  uint32_t start;
+  if (!args[2]->Uint32Value(ctx).To(&start)) return;
+  uint32_t end;
+  if (!args[3]->Uint32Value(ctx).To(&end)) return;
```

Comment:
:thinking: shouldn't this then be semver-major to avoid possible breakage?

Also, perhaps we can
```c++
  if (!args[2]->IsUint32() || !args[3]->IsUint32())
    return args.GetReturnValue().Set(-2);

  uint32_t start = args[2].As<Uint32>()->Value();
  uint32_t end = args[3].As<Uint32>()->Value();
```
in this case (tests seem to pass)?

Reply:
The only place where we pass in invalid arguments is from a `process.binding()` test. We could/should probably remove that test in a follow-up PR, and revert 05aa50f5fb6c272dc20e0d6362b1aabb2fe4e94e (which is pretty close to your suggestion)

Reply:
Oh, I thought that this was expected somewhere else, therefore the test. The suggested commit is even better, with the above I wanted to somehow mitigate that test.